### PR TITLE
Improvement for separate apps

### PR DESCRIPTION
**Pull request by D4nte**
Helps with comit-network/create-comit-app#67

### CODE REVIEW

File: src/comitClient.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export class ComitClient {
   constructor(
     private readonly bitcoinWallet: BitcoinWallet,
     private readonly ethereumWallet: EthereumWallet,
-    private readonly cnd: Cnd
+    public readonly cnd: Cnd
```
Comment (luckysori) [inline src/comitClient.ts]: :question: I think you are only doing this to get the peer ID and the address hint. What if `ComitClient` handled getting this for you instead?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+- `Swap.getEntity()` to get details of the swap.
+
+### Changed
+- Replace Siren types with Expect Swap types to improve interfaces accuracy.
```
Comment (luckysori) [inline CHANGELOG.md]:
```suggestion
- Rename `SwapEntity` to `SwapProperties`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+- `Swap.getEntity()` to get details of the swap.
```
Comment (luckysori) [inline CHANGELOG.md]:
```suggestion
- `Swap.getEntity()` to get details of the swap.
- Export `SwapEntity` and `SwapSubEntity`.
```

File: dist/comitClient.d.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { Swap } from "./swap";
 export declare class ComitClient {
     private readonly bitcoinWallet;
     private readonly ethereumWallet;
-    private readonly cnd;
+    readonly cnd: Cnd;
```
[the remark this answers is not in the document]
Comment (D4nte) [reply dist/comitClient.d.ts]: Yes the motivation is to access the peer id and address hint.

Please propose a way forward.

File: src/swap.ts [hunk 5]
```diff
@@ -45,6 +45,11 @@ export class Swap {
     return await this.doLedgerAction(response.data);
   }
 
+  public async getEntity(): Promise<SwapEntity> {
```
[the remark this answers is not in the document]
Comment (D4nte) [reply src/swap.ts]: Currently it is not possible to get any details from the swap we are talking about! See how `getEntity` is used in #129: This is to know what are the alpha/beta assets and their quantities.

Please propose a way forward.

File: dist/cnd.d.ts [hunk 6]
```diff
@@ -58,7 +58,13 @@ export declare type LedgerAction = {
...
     payload: EthereumCallContractPayload;
 };
-export interface SwapEntity {
+export interface SwapEntity extends Entity {
+    properties?: SwapProperties;
+}
+export interface SwapSubEntity extends EmbeddedRepresentationSubEntity {
+    properties?: SwapProperties;
+}
+export interface SwapProperties {
```
[the remark this answers is not in the document]
Comment (D4nte) [reply dist/cnd.d.ts]: I don't think it's a breaking change. I replace the original return type `A` by a new return type `B` that `extends A` (and hence has the same properties).

File: src/comitClient.ts (hunk 1 above)
Comment (D4nte) [reply src/comitClient.ts]: Ok.

**Comment by D4nte** [comment]
Note to self: rebase before merging.

File: dist/comitClient.d.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (D4nte) [reply dist/comitClient.d.ts]: I'll do @luckysori's suggestion here.

**Comment by luckysori** [comment]
> > Sorry.
> 
> Instead of apologising could you please specify what is blocking for you on this PR?

The only thing that was marked with a :mega:, as is customary.

**Comment by D4nte** [comment]
> > > Sorry.
> > 
> > 
> > Instead of apologising could you please specify what is blocking for you on this PR?
> 
> The only thing that was marked with a 📣, as is customary.

Ah yes indeed, as I processed this comment first my brain just dismissed it. I am the one sorry ;)

File: src/swap.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (da-kami) [reply src/swap.ts]: Summary: 
`Cnd` should be abstracted away, the SDK should only expose the `ComitClient` for communication with `Cnd`. 

For now: move on. Create follow-up PR if needed.

**Pull request merged**